### PR TITLE
Create and update records in question's new tables

### PR DESCRIPTION
**Pull request by guillogo**

### CODE REVIEW

**Review by safatshahin — COMMENTED** [review]
Hi @guillogo 
There are some unit tests failing which are related to questiontype base and the question generator. Can you please have a look if they can be fixed in this branch?

Cheers

**Comment by safatshahin** [comment]
Hi @guillogo 
Sorry, just to add to my previous comment, this issue should include the changes to the core_question unit test generator, which handles the creation of data.
Cheers!

**Comment by guillogo** [comment]
Hi @safatshahin 

Thank you, can you have a look please: My comments are here: https://github.com/catalyst/moodle-MDL-70329/issues/189#issuecomment-893897963

Thank you.

**Review by safatshahin — COMMENTED** [review]
Hi @guillogo 
Thanks for working on this, I have made some comments around methods to questionlib as well as some branching changes. As soon as they are done, it's good to go.
Cheers!

File: question/type/questiontypebase.php [hunk 1]
```diff
@@ -486,10 +504,6 @@ public function save_question($question, $form) {
                     '$result->noticeyesno no longer supported in save_question.');
         }
 
-        // Give the question a unique version stamp determined by question_hash().
-        $DB->set_field('question', 'version', question_hash($question),
```
[the remark this answers is not in the document]
Comment (guillogo) [reply question/type/questiontypebase.php:490]: done, deprecated without replacement. Added comment in upgrade.txt too.

File: question/engine/tests/helpers.php [hunk 2]
```diff
@@ -200,7 +199,6 @@ public static function initialise_question_data($qdata) {
         $qdata->penalty = 0.3333333;
         $qdata->length = 1;
         $qdata->stamp = make_unique_id_code();
-        $qdata->version = make_unique_id_code();
```
[the remark this answers is not in the document]
Comment (guillogo) [reply question/engine/tests/helpers.php:203]: Hi, the make_unique_id_code is a moodlelib function so I'll leave it.

**Comment by guillogo** [comment]
Hi @safatshahin changes done. Thank you.

**Pull request merged**